### PR TITLE
fix: avoid empty where statement in queries

### PR DESCRIPTION
**Pull request by defudef**
### Related Issues

closes #

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by defudef** [comment]
Fixes following issue (`where` as an empty string):

image

Which caused an error:

```
Malformed parameter: where: Syntax error while parsing 'where'. Unexpected end of input, expected not or conditional expression (line 1, column 1)
```

File: packages/commercetools/api-client/src/helpers/search/index.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ const buildProductWhere = (search: ProductSearch) => {
     ];
   }
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/helpers/search/index.ts:81]: exit earlier
```
  if (predicates.length === 0) {
    return null;
  }
```

File: packages/commercetools/api-client/src/helpers/search/index.ts [hunk 2]
```diff
@@ -91,15 +91,15 @@ const buildCategoryWhere = (search: CategorySearch) => {
     return `slug(${locale}="${search.slug}")`;
   }
 
-  return '';
+  return undefined;
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/helpers/search/index.ts:94]: null works here

File: packages/commercetools/api-client/src/helpers/search/index.ts [hunk 3]
```diff
@@ -91,15 +91,15 @@ const buildCategoryWhere = (search: CategorySearch) => {
...
 };
 
 const buildOrderWhere = (search: OrderSearch): string => {
   if (search?.id) {
     return `id="${search.id}"`;
   }
 
-  return null;
+  return undefined;
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/helpers/search/index.ts:102]: same here

**Comment by bloodf** [comment]
This is closed due to inactivity.

If you want your PR to be re-opened, please update your files, check for any conflicts.

If your PR is for Vue Storefront 1 target the new repository at https://github.com/vuestorefront/vue-storefront-1

**Pull request closed without merging**